### ISSUE TITLE
Tramp support

### ISSUE THREAD

**Issue by egh** [kubernetes-el/kubernetes-el#165]
It would be useful (maybe?) to have access to tramp to view files in a container.

Here is a working example:
```elisp
  (defun kubernetes-tramp--running-containers (&optional _ignored)
    "A tramp-completion function for kubernetes."
    (if (null (kubernetes-state))
        (kubernetes-pods-refresh-now))
    (-let (((&alist 'items pods) (kubernetes-state-pods (kubernetes-state))))
      (-map (-lambda ((&alist 'metadata (&alist 'name)))
              (list nil name))
            pods)))
  (add-to-list 'tramp-methods
               `("kubectl"
                 (tramp-login-program ,kubernetes-kubectl-executable)
                 (tramp-login-args (("exec" "-i" "-t") ("-u" "%u") ("%h") ("--" "sh")))
                 (tramp-remote-shell "/bin/sh")
                 (tramp-remote-shell-args ("-i" "-c"))))
  (tramp-set-completion-function "kubectl" '((kubernetes-tramp--running-containers "")))
```

As an example of similar functionality, the https://github.com/Silex/docker.el package allows one to dired into a volume, which is quite useful.

https://github.com/Silex/docker.el/blob/master/docker-volume.el#L75

**Comment by noorul** [comment]
Wow!
@egh Will you be interested to create a PR ?

**Comment by noorul** [comment]
I was thinking about this yesterday.

**Comment by egh** [comment]
I'm happy to create a PR. I'm curious if you have any advice about the commands that might be useful here. For instance, would it be useful to have a command that tried to open a `dired` buffer on the current pod? If so, what would the key be? I think `d` and `D` are taken.

**Comment by noorul** [comment]
If we follow what `docker` did, then I think `find-file` is an apt workflow. What do you think?

**Comment by noorul** [comment]
@egh I left a couple of review comments for the PR.

**Comment by jinnovation** [comment]
Closed by #167. Thanks for this.
